RIOT-OS/RIOT
# cpu/samd5x: add SD Host Controller implementation

### Contribution description

This implements support for the SD/MMC host controller found on SAM D5x/E5x family chips.
It is based on the code from [initmaker](https://github.com/alkgrove/initmaker) (and the accompanying [blog post](http://www.alkgrove.com/articles/SAMD51_MicroSD_article.html)) with interrupt support instead of busy-waiting.




### Testing procedure

Insert an SD card into the SD card slot of the `same54-xpro`

`tests/mtd_raw` should work:

```
2022-03-25 11:57:43,168 # main(): This is RIOT! (Version: 2022.04-devel-912-g8732f-sam0_sdhc)
2022-03-25 11:57:43,169 # Manual MTD test
2022-03-25 11:57:43,173 # init MTD_0… OK (32 MiB)
2022-03-25 11:57:43,176 # init MTD_1… OK (256 byte)
2022-03-25 11:57:43,210 # init MTD_2… OK (7580 MiB)
> test 2
2022-03-25 11:57:44,557 # test 2
2022-03-25 11:57:44,558 # [START]
2022-03-25 11:57:44,588 # [SUCCESS]
```

We have an internal benchmark tool that generates data records to and stores them on the SD card, there we see improvements compared to `sdcard_spi`:

#### sdcard_spi

```
2022-03-25 17:30:01,204 # bench 1000
2022-03-25 17:30:06,203 # write:   4991646us  ---  4991.646us per call  ---        200 calls per sec
```

#### sam0_sdhc

```
2022-03-25 12:42:14,891 # bench 1000
2022-03-25 12:42:17,130 # write:   2230878us  ---  2230.878us per call  ---        448 calls per sec
```

### Issues/PRs references
